PhysicsTools/PatAlgos/python/jetCollectionTools.py

```diff
@@ -0,0 +1,621 @@
+    #
+    genFlavourTable = "{}FlavourTable".format(tagName)
+    if genFlavourTable in self.main:
+      raise ValueError("Step '%s' already implemented" % genFlavourTable)
+    setattr(proc, genFlavourTable, cms.EDProducer("GenJetFlavourTableProducer",
+        name            = cms.string(name),
+        src             = cms.InputTag(genJetsCollection),
+        cut             = cms.string(""),
+        deltaR          = cms.double(0.1),
+        jetFlavourInfos = cms.InputTag(genFlavour),
+      )
+    )
+    currentTasks.append(genFlavourTable)
+
+    self.main.extend(currentTasks)
+
+#============================================
+#
+# RecoJetInfo
+#
+#============================================
+class RecoJetInfo(object):
+  def __init__(self, jet, inputCollection):
+    self.jet = jet
+    self.inputCollection = inputCollection
+    algoKey     = 'algo'
+    sizeKey     = 'size'
+    recoKey     = 'reco'
+    puMethodKey = 'puMethod'
+    jetRegex = re.compile(
+      r'(?P<{algo}>({algoList}))(?P<{size}>[0-9]+)(?P<{reco}>(pf|calo))(?P<{puMethod}>(chs|puppi|sk|cs|))'.format(
+        algo     = algoKey,
+        algoList = '|'.join(supportedJetAlgos.keys()),
+        size     = sizeKey,
+        reco     = recoKey,
+        puMethod = puMethodKey,
+      )
+    )
+    jetMatch = jetRegex.match(jet.lower())
+    if not jetMatch:
+      raise RuntimeError('Invalid jet collection: %s' % jet)
+
+    self.jetAlgo     = jetMatch.group(algoKey)
+    self.jetSize     = jetMatch.group(sizeKey)
+    self.jetReco     = jetMatch.group(recoKey)
+    self.jetPUMethod = jetMatch.group(puMethodKey)
+
+    self.jetSizeNr = float(self.jetSize) / 10.
+
+    self.doCalo = self.jetReco == "calo"
+    self.doCS   = self.jetPUMethod == "cs"
+    self.skipUserData = self.doCalo or (self.jetPUMethod in [ "puppi", "sk" ] and inputCollection == "")
+    
+    self.jetCorrPayload = "{}{}{}".format(
+      self.jetAlgo.upper(), self.jetSize, "Calo" if self.doCalo else self.jetReco.upper()
+    )
+    if self.jetPUMethod == "puppi":
+      self.jetCorrPayload += "Puppi"
+    elif self.jetPUMethod in [ "cs", "sk" ]:
+      self.jetCorrPayload += "chs"
+    else:
+      self.jetCorrPayload += self.jetPUMethod.lower()
+      
+#============================================
+#
+# RecoJetAdder
+#
+#============================================
+class RecoJetAdder(object):
+  def __init__(self):
+    self.prerequisites = []
+    self.main = []
+    self.bTagDiscriminators = [
+      'pfTrackCountingHighEffBJetTags',
+      'pfTrackCountingHighPurBJetTags',
+      'pfJetProbabilityBJetTags',
+      'pfJetBProbabilityBJetTags',
+      'pfSimpleSecondaryVertexHighEffBJetTags',
+      'pfSimpleSecondaryVertexHighPurBJetTags',
+      'pfCombinedSecondaryVertexV2BJetTags',
+      'pfCombinedInclusiveSecondaryVertexV2BJetTags',
+      'pfCombinedMVAV2BJetTags',
+      'pfDeepCSVJetTags:probb',
+      'pfDeepCSVJetTags:probbb',
+      'pfBoostedDoubleSecondaryVertexAK8BJetTags',
```

Comment:
is this some ad hoc list or is this also defined somewhere else?
If it is supposed to match another place, better get the full list from there.